gnucash/register/ledger-core/gnc-ledger-display.c

```diff
@@ -935,11 +909,38 @@ gnc_ledger_display_refresh (GNCLedgerDisplay* ld)
         return;
     }
 
+    /* if subaccount ledger, check to see if still the same number
+     * of subaccounts, if not recreate the query. */
+    if (ld->ld_type == LD_SUBACCOUNT)
+    {
+        Account* leader = gnc_ledger_display_leader (ld);
+        GList* accounts = gnc_account_get_descendants (leader);
+
+        if (g_list_length (accounts) != ld->number_of_subaccounts)
+            gnc_ledger_display_make_query (ld,
+                                        gnc_prefs_get_float (GNC_PREFS_GROUP_GENERAL_REGISTER, GNC_PREF_MAX_TRANS),
+                                        gnc_get_reg_type (leader, ld->ld_type));
+
+        g_list_free (accounts);
+    }
+
+    /* In lieu of not "mis-using" some portion of the infrastructure by writing
+     * a bunch of new code, we just filter out the accounts of the template
+     * transactions.  While these are in a separate Account trees just for this
+     * reason, the query engine makes no distinction between Account trees.
+     * See Gnome Bug 86302.
+     *         -- jsled */
     // Exclude any template accounts for search register and gl
     if (!ld->reg->is_template && (ld->reg->type == SEARCH_LEDGER || ld->ld_type == LD_GL))
         exclude_template_accounts (ld->query, ld->excluded_template_acc_hash);
 
-    gnc_ledger_display_refresh_internal (ld, qof_query_run (ld->query));
+    /* It's not clear if we should re-run the query, or if we should
+     * just use qof_query_last_run().  It's possible that the dates
+     * changed, requiring a full new query.  Similar considerations
+     * needed for multi-user mode.
+     */
```

Comment:
This comment goes with what is now line 880.

Reply:
I've moved it now